terraform-beyond-layer0/example-2/main.tf

```diff
@@ -0,0 +1,91 @@
+#
+# ./example2/main.tf
+#
+
+provider "layer0" {
+  endpoint        = "${var.endpoint}"
+  token           = "${var.token}"
+  skip_ssl_verify = true
+}
+
+provider "aws" {
+  access_key = "${var.access_key}"
+  secret_key = "${var.secret_key}"
+  region     = "${var.region}"
+}
+
+# create one layer0 environment for each map value defined in variable 'environments'
+resource "layer0_environment" "demo" {
+  count = "${length(var.environments)}"
+
+  name = "${lookup(var.environments, count.index)}_demo"
```

Comment:
map -> list ✅ 

re: [nit] not sure about this one. `_demo` suffix I anticipate would be for name of a application. I.e., if I had multiple applications, I anticipate environments like:

- dev_application1
- staging_application1
- dev_application2
- staging_application2